abe/helper_functions/mongodb_helpers.py

```diff
@@ -0,0 +1,31 @@
+from functools import wraps
+
+import mongoengine
+
+# This is sketchy, but mongoengine doesn't export a helpful base class that
+# could be used to catch all and only input-related errors (as opposed to server
+# and implementation errors.)
+#
+# If this breaks with a future release, could be released by an extensive list
+# of the relevant errors.
+mongo_errors = tuple(mongoengine.errors.__dict__[k] for k in mongoengine.errors.__all__)
+
+
+def mongo_resource_errors(f):
```

Comment:
Not sure about the name.

Also, maybe this should be middleware around the whole app. Or, decorate at the resource level (if this is possible) instead of decorating each route?